### ISSUE TITLE
new provisinoing engine: web parts get doubled on the page

### ISSUE THREAD

**Issue by maratbakirov** [OfficeDev/PnP#772]
When I do create a page and webparts inside

 <Files>
   <File Src="Home.aspx" Folder="Pages" Overwrite="true">
     <Properties>
       <Property Key="Title" Value="Home" />
       
       <Property Key="ContentTypeId" Value="0x010100C568DB52D9D0A14D9B2FDCC96666E9F2007948130EC3DB064584E219954237AF390064DEA0F50FC8C147B0B6EA0636C4A7D4" />
       <Property Key="PublishingPageLayout" Value="~sitecollection/_catalogs/masterpage/BlankWebPartPage.aspx, Blank Web Part Page" />
     </Properties>
   </File>
<WebParts>
 ....
</WebParts>
 </Files>

and run the code twice, the web parts get doubled.

basically each time you run the code the web parts are added again. 

this fie 

PnP-dev\OfficeDevPnP.Core\OfficeDevPnP.Core\Framework\Provisioning\ObjectHandlers\ObjectFiles.cs

should have a check for that 

This is because there is no code in the objectfilehandler that would check if the web part with this title already exists on the page.

**Comment by erwinvanhunen** [comment]
Good one! Fixed in PR #775 . Closing the issue.
